### PR TITLE
Panzer: allow HVol elements

### PR DESCRIPTION
**Pull request by egphill**
Adapted CurlLaplacianExample into a mixed version that uses both HCurl and HDiv elements in 3D or HCurl and HVol elements in 2D.

Exposed HVol basis type and defined its basis values. Modified L2Projection to allow HVol elements.

@trilinos/<teamName>

## Description

Previously HVol elements were allowed in Panzer only through the Const basis option. This was restricted to zeroth order and did not use the HVOL basis value transformation for defining the basis. This change keeps the Const basis but adds the option to use arbitrary order HVol elements on 2D meshes.

## Motivation and Context

The z component of an HDiv DOF should use an HVol basis in 2D. This is necessary for arbitrary order electromagnetics in 2D.

## How Has This Been Tested?

Added the test example MixedCurlLaplacianExample. This is the same test as CurlLaplacianExample except it's in mixed form such that it uses both HCurl elements for the primary DOF and HVol elements for the curl of that DOF.

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

Most everything in the MixedCurlLaplacianExample directory is copied directly from CurlLaplacianExample. The changes of interest there are in Example_CurlLaplacianEquationSet_impl.hpp, main.cpp, and CMakeLists.txt.

### CODE REVIEW

**Comment by egphill** [comment]
@mperego I want to keep Const for backward compatibility first of all. But in general, it might be useful. It's a piecewise constant basis that isn't scaled by the volume of the cell, so the basis is just one on every element.

File: packages/panzer/adapters-stk/example/MixedCurlLaplacianExample/Debugging.nb [hunk 1]
```diff
@@ -0,0 +1,1741 @@
+(* Content-type: application/vnd.wolfram.mathematica *)
```
Comment (rppawlo) [inline packages/panzer/adapters-stk/example/MixedCurlLaplacianExample/Debugging.nb]: did you mean to commit this file?

**Comment by mperego** [comment]
I see. 
Is Panzer using HVol for fields that are divergence of HDiv fields (in 2d and 3d) and curls of Hcurl fields in 2d?
I just want to make sure that we don't use the const basis when we should use the HVol basis instead.

**Comment by egphill** [comment]
@mperego Now that we have HVol, we should be using it how you say. Historically, application codes have used Const for curls of HCurl fields in 2D, and it's been working at lowest order.

File: packages/panzer/adapters-stk/example/MixedCurlLaplacianExample/Debugging.nb (hunk 1 above)
Comment (egphill) [reply packages/panzer/adapters-stk/example/MixedCurlLaplacianExample/Debugging.nb]: I can remove this file.

**Comment by egphill** [comment]
@rppawlo I did not add an HVol projection test. Would you like me to?

**Comment by rppawlo** [comment]
If it's quick, we probably should. I have a 2D projection test sitting on a branch so we could wait until that is merged in and add it there. If you wan to push now that's fine too.

**Comment by egphill** [comment]
@rppawlo Is the 2D projection test you're talking about a 2D version of the ToNodal test in the L2Projection tests? If so, I think it'd be easiest just to include the HVol test there.

**Comment by rppawlo** [comment]
> Is the 2D projection test you're talking about a 2D version of the ToNodal test in the L2Projection tests? If so, I think it'd be easiest just to include the HVol test there.

yes. I'll get that pushed next week and add it there.

File: packages/panzer/adapters-stk/example/MixedCurlLaplacianExample/Debugging.nb (hunk 1 above)
Comment (egphill) [reply packages/panzer/adapters-stk/example/MixedCurlLaplacianExample/Debugging.nb]: Ok, I removed it. This needs to be re-approved.

**Pull request merged**